src/sql/azdata.proposed.d.ts

```diff
@@ -740,6 +740,10 @@ declare module 'azdata' {
 		flavor: string;
 	}
 
+	export interface QueryExecutionOptions {
+		options: Map<string, any>;
```

Comment:
Why does this need to be an entire struct? can't the argument passed to the function just be a Map?

Reply:
Having a class that encapsulates the options works better with the backend deserializer.

Reply:
We probably could remove this from the extensibility API parameters and only use the wrapper class when communicating with the DMP libraries.  i.e. change the signature for `setExecutionOptions`.  I'll take a look if that cleans up the extension sample code a bit.

Reply:
I updated the extension-facing API to move this class from the parameters list.